### PR TITLE
Fix incompatible box props

### PR DESCRIPTION
**Pull request by jonotrujillo**
**What:**

- Avoid collision of `color` prop
- Add default theme declaration
- Use theme declaration in `Box` component
​
**Why:**

- Fix issue #190 
- Fix another issue where `Box` requires a value for `theme`

**Checklist:**

-   [x] Ready to be merged

### CODE REVIEW

File: src/components/Box/Box.tsx [hunk 1]
```diff
@@ -24,13 +24,13 @@ interface BoxProps
     extends SpaceProps,
         LayoutProps,
         PositionProps,
-        ColorProps,
+        Omit<ColorProps, 'color'>, // HACK: avoid collision of `color` prop
```
Comment (nlopin) [inline src/components/Box/Box.tsx:27]: what else do we have inside colorProps in styled-system?

File: src/components/Box/Box.tsx [hunk 2]
```diff
@@ -24,13 +24,13 @@ interface BoxProps
...
         FlexboxProps,
         GridProps,
         BackgroundProps,
         TextAlignProps {}
 
-const Box: StyledComponentBase<'div', never, BoxProps, 'theme'> = styled.div.attrs({ theme })<BoxProps>`
+const Box: StyledComponentBase<'div', DefaultTheme, BoxProps, 'theme'> = styled.div.attrs({ theme })<BoxProps>`
```
Comment (nlopin) [inline src/components/Box/Box.tsx:33]: Have you tried stripping off the whole `StyledComponentBase` type? I think Typescript can derive the correct type from the assignment

**Review by div-Leo — APPROVED** [review]

File: src/components/Box/Box.tsx (hunk 2 above)
Comment (jonotrujillo) [reply src/components/Box/Box.tsx:33]: Then the project won't compile, because of other component that depend on `Box` breaking. I would like to have it this way, though. Looking into it now.

File: src/components/Box/Box.tsx (hunk 1 above)
Comment (jonotrujillo) [reply src/components/Box/Box.tsx:27]: bg color and opacity

File: src/components/Box/Box.tsx (hunk 2 above)
Comment (jonotrujillo) [reply src/components/Box/Box.tsx]: This is done now in c525f49 and 0405c22

File: src/components/Box/Box.spec.tsx [hunk 3]
```diff
@@ -33,4 +33,11 @@ describe('Box', () => {
             render(<Box gridColumnGap={2} gridTemplateRows="1fr auto 10%" />).container.firstChild
         ).toMatchSnapshot();
     });
+
+    it('accepts props spreading', () => {
+        const renderBox = (props: BoxProps) => <Box {...props} />;
+        expect(
+            render(renderBox({ backgroundColor: Colors.POSITIVE_GREEN_900 })).container.firstChild
+        ).toMatchSnapshot();
```
Comment (nlopin) [inline src/components/Box/Box.spec.tsx:41]: What is the idea behind making a snapshot? if the test checks prop spreading it should test only that the code is not crashing, no? 

Do you think we can remove the snapshot?

File: src/components/Search/Search.tsx [hunk 4]
```diff
@@ -32,7 +34,11 @@ const ActiveBox = styled(Box)`
     }
 `;
 
-const SearchInputContainer = styled(Box)`
```
Comment (nlopin) [inline src/components/Search/Search.tsx:35]: I'm not quite getting why this change is here

**Review by nlopin — APPROVED** [review]
My main concern are snapshot test, let's discuss and merge

File: src/components/Box/Box.spec.tsx (hunk 3 above)
Comment (jonotrujillo) [reply src/components/Box/Box.spec.tsx:41]: No need for the snapshot, I can change the assertion to `not.toThrow()`. What do you think?

File: src/components/Search/Search.tsx (hunk 4 above)
Comment (jonotrujillo) [reply src/components/Search/Search.tsx:35]: This changed is needed after removing the `StyledComponentBase` typing from `Box`. As `isInFocus` is not a prop in `BoxProps`, the interface is extended here.
